java/sage/IOUtils.java

```diff
@@ -1405,18 +1408,64 @@ public static String convertSMBURLToUNCPath(String smbPath)
     return sb.toString();
   }
 
-  // Creates a java.io.BufferedReader for the specified file and checks the first two bytes for the Unicode BOM and sets the
-  // charset accordingly; otherwise if there's no BOM it'll use the passed in defaultCharset
+  /**
+   * Creates a java.io.BufferedReader for the specified file and checks the first two bytes for the
+   * Unicode BOM and sets the charset accordingly; otherwise if there's no BOM it'll use the passed
+   * in defaultCharset
+   *
+   * @param filePath The file to be opened.
+   * @param defaultCharset The charset to be used if the BOM is missing.
+   * @return A BufferedReader.
+   * @throws java.io.IOException If an I/O error occurs.
+   */
   public static java.io.BufferedReader openReaderDetectCharset(String filePath, String defaultCharset) throws java.io.IOException
   {
-    return openReaderDetectCharset(new java.io.File(filePath), defaultCharset);
+    return openReaderDetectCharset(new java.io.File(filePath), defaultCharset, true);
   }
+
+  /**
+   * Creates a java.io.BufferedReader for the specified file and checks the first two bytes for the
+   * Unicode BOM and sets the charset accordingly; otherwise if there's no BOM it'll use the passed
+   * in defaultCharset
+   *
+   * @param filePath The file to be opened.
+   * @param defaultCharset The charset to be used if the BOM is missing.
+   * @return A BufferedReader.
+   * @throws java.io.IOException If an I/O error occurs.
+   */
   public static java.io.BufferedReader openReaderDetectCharset(java.io.File filePath, String defaultCharset) throws java.io.IOException
   {
-    java.io.FileInputStream fis = null;
+    return openReaderDetectCharset(filePath, defaultCharset, true);
+  }
+
+  /**
+   * Creates a java.io.BufferedReader for the specified file and checks the first two bytes for the
+   * Unicode BOM and sets the charset accordingly; otherwise if there's no BOM it'll use the passed
+   * in defaultCharset
+   *
+   * @param filePath The file to be opened.
+   * @param defaultCharset The charset to be used if the BOM is missing.
+   * @param local Pass true if the file is locally accessible. If this value is false, the
+   *              BufferedReader will be backed by a MediaServerInputStream.
+   * @return A BufferedReader.
+   * @throws java.io.IOException If an I/O error occurs.
+   */
+  public static java.io.BufferedReader openReaderDetectCharset(java.io.File filePath, String defaultCharset, boolean local) throws java.io.IOException
+  {
+    java.io.InputStream fis = null;
     try
     {
-      fis = new java.io.FileInputStream(filePath);
+      if (local)
+        fis = new java.io.FileInputStream(filePath);
+      else
+      {
+        // This is crucial. If we don't do this step the file will almost certainly be inaccessible.
+        sage.NetworkClient.getSN().requestMediaServerAccess(filePath, true);
```

Comment:
Can you add a finally{} clause which released the media server access so the HashSet on the server doesn't grow without bound?